### PR TITLE
Move compile spec to ArmTester interface

### PR DESCRIPTION
**Pull request by freddan80**
* Create compile spec builder
* Added default compile spec for unit tests
* Cleaned up some redundant parameters

### CODE REVIEW

**Comment by freddan80** [comment]
CI failure unrelated to this PR

**Comment by freddan80** [comment]
CI fails not related to this PR

File: backends/arm/arm_backend.py [hunk 1]
```diff
@@ -38,6 +38,118 @@
...
 
+class ArmCompileSpecBuilder:
+    def __init__(self):
+        self.compile_spec: List[CompileSpec] = []
+        self.compiler_flags = []
+        self.output_format = None
+        self.path_for_intermediates = None
+        self.permute_nhwc = False
+
+    def ethosu_compile_spec(
```
Comment (digantdesai) [inline backends/arm/arm_backend.py:49]: why not just do this in `__init__` or you can create a `@classmethod` if you want to write another constructor.

File: backends/arm/arm_backend.py [hunk 2]
```diff
@@ -38,6 +38,118 @@
...
+        """
+        Generate compile spec for TOSA flatbuffer output
+        """
+        assert (
+            self.output_format is None
+        ), f"Output format already set: {self.output_format}"
+        self.output_format = "tosa"
+        return self
+
+    def dump_intermediate_tosa(self, output_path: str):
```
Comment (digantdesai) [inline backends/arm/arm_backend.py:96]: can all of these be `@property`? 

So you can just do 

`builder.dump_intermediate_tosa = True`

and 

`if builder.dump_intermediate_tosa`

**Review by digantdesai — APPROVED** [review]
LGTM

**Review by digantdesai — APPROVED** [review]
LGTM

**Review by mergennachin — APPROVED** [review]

**Pull request closed without merging**